airflow/ui/src/pages/DagsList/DagsFilters.tsx

```diff
@@ -51,6 +57,21 @@ export const DagsFilters = () => {
       [pagination, searchParams, setSearchParams, setTableURLState, sorting],
     );
 
+  const handleStateChange: React.MouseEventHandler<HTMLButtonElement> =
```

Comment:
Why a `React.MouseEventHandler` but `React.MouseEventHandler` above for the `React.MouseEventHandler`?

Reply:
I initially had ChangeEventHandler but got below error and changed back to MouseEventHandler as suggested by the linter to keep it compatible with `onClick` .

```
const handleStateChange: React.ChangeEventHandler<HTMLButtonElement> =
```

Lint error : 

```
src/pages/DagsList/DagsFilters.tsx:85:15 - error TS2322: Type 'ChangeEventHandler<HTMLButtonElement>' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
  Types of parameters 'event' and 'event' are incompatible.
    Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'ChangeEvent<HTMLButtonElement>'.
      Types of property 'target' are incompatible.
        Type 'EventTarget' is not assignable to type 'EventTarget & HTMLButtonElement'.
          Type 'EventTarget' is missing the following properties from type 'HTMLButtonElement': disabled, form, formAction, formEnctype, and 311 more.

85               onClick={handleStateChange}
                 ~~~~~~~

  node_modules/.pnpm/@types+react@18.3.5/node_modules/@types/react/index.d.ts:2489:9
    2489         onClick?: MouseEventHandler<T> | undefined;
                 ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & ButtonProps'
```

Reply:
Paused is a select which is a ChangeEvent while clicking on a button is just a MouseEvent

Reply:
Got it thanks.